### ISSUE TITLE
Linux bower install fail 

### ISSUE THREAD

**Issue by gem874** [eirslett/frontend-maven-plugin#779]
Hello, I have a question. Please help me!
I want to run a project on Linux but it did not work.
I saw a similar problem and tried it. However, I saw the message "Permission deny".
* https://github.com/eirslett/frontend-maven-plugin/issues/216

P.s. Window execution works well.

1. pom.xml 
<build>
		<plugins>
			<plugin>
				<groupId>org.springframework.boot</groupId>
				<artifactId>spring-boot-maven-plugin</artifactId>
			</plugin>

			
			<plugin>
				<groupId>com.github.eirslett</groupId>
				<artifactId>frontend-maven-plugin</artifactId>
				<version>1.6</version>
				
				<configuration>
					<installDirectory>${project.basedir}</installDirectory>
					<workingDirectory>src/main/webapp/front-dev</workingDirectory>
					<npmInheritsProxyConfigFromMaven>false</npmInheritsProxyConfigFromMaven>
					<bowerInheritsProxyConfigFromMaven>false</bowerInheritsProxyConfigFromMaven>
				</configuration>
				<executions>
					<execution>
						<id>install node and npm</id>
						<goals>
							<goal>install-node-and-npm</goal>
						</goals>
						<configuration>
							<nodeVersion>v8.11.1</nodeVersion>
							<npmVersion>5.6.0</npmVersion>
						</configuration>
					</execution>

					<execution>
						<id>npm Install</id>
						<goals>
							<goal>npm</goal>
						</goals>
						<configuration>
							<arguments>install</arguments>
						</configuration>
					</execution>

					<execution>
						<id>bower Install</id>
						<goals>
							<goal>npm</goal>
						</goals>
						<configuration>
							<arguments>install -g bower</arguments>
						</configuration>
					</execution>
				</executions>
			</plugin>
			
		</plugins>
		<resources>
			<resource>
				<directory>${project.build.sourceDirectory}</directory>
				<includes>
					<include>**/*.xml</include>
				</includes>
			</resource>
		</resources>
	</build>

1-2. profile setting 

> 						<groupId>com.github.eirslett</groupId>
						<artifactId>frontend-maven-plugin</artifactId>
						<version>1.6</version>
						
						<configuration>
							<installDirectory>${project.basedir}</installDirectory>
							<workingDirectory>src/main/webapp/front-dev</workingDirectory>
							<npmInheritsProxyConfigFromMaven>false</npmInheritsProxyConfigFromMaven>
							<bowerInheritsProxyConfigFromMaven>false</bowerInheritsProxyConfigFromMaven>
						</configuration>

2.  npm, bower script file ( in my work space) 

`#!/bin/sh
PATH="$PWD/node/":$PATH
node "node/node_modules/npm/bin/npm-cli.js" "$@"`

`#!/bin/sh
"node/node" "node_modules/bower/bin/bower" "$@"`

3. When I build in jenken, I got a message 
`[INFO] > bower install
[INFO] 
[ERROR] sh: ./bower: Permission denied
[ERROR] npm ERR! code ELIFECYCLE
[ERROR] npm ERR! errno 126
[ERROR] npm ERR! NGS@1.0.0 bower: `bower install`
[ERROR] npm ERR! Exit status 126
[ERROR] npm ERR! 
[ERROR] npm ERR! Failed at the NGS@1.0.0 bower script.
[ERROR] npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
[ERROR] `

**Comment by eirslett** [comment]
Yup, don't do `install -g bower`, install it locally instead, without the `-g`.
But really though - most people use npm nowadays, I recommend that you switch to it.

**Comment by gem874** [comment]
hello! thank you for answer! 
I change like this ! 

1. pom.xml 
.... 
<execution>
						<id>npm Install</id>
						<goals>
							<goal>npm</goal>
						</goals>
						<configuration>
							<arguments>install</arguments>
						</configuration>
					</execution>

					<execution>
						<id>bower Install</id>
						<goals>
							<goal>npm</goal>
						</goals>
						<configuration>
							<arguments>install bower</arguments>
						</configuration>
					</execution>


.... 



2. change file npm, bower  
(Path is diffrent, because I change working directory, so I change node path) 

#!/bin/sh
PATH="$PWD/node/":$PATH
node "../../../../node/node_modules/npm/bin/npm-cli.js" "$@"


#!/bin/sh
"../../../../node/node" "../../../../node_modules/bower/bin/bower" "$@"
